closure/goog/array/array.js

```diff
@@ -496,8 +496,8 @@ goog.array.count = function(arr, f, opt_obj) {
  *     for every element. This function takes 3 arguments (the element, the
  *     index and the array) and should return a boolean.
  * @param {S=} opt_obj An optional "this" context for the function.
- * @return {T|null} The first array element that passes the test, or null if no
- *     element is found.
+ * @return {T|string|null} The first array element that passes the test, or
```

Comment:
Presumably the function is a function(string, number) which should set T to string already, no?

Also, it seems like making String implement IArrayLike<string> would maybe be a better way to go about fixing these issues.